core/src/traits/block_storage_interface.rs

```diff
@@ -1,19 +1,19 @@
 use crate::structures::produced_block::ProducedBlock;
 use anyhow::Result;
 use async_trait::async_trait;
-use solana_rpc_client_api::config::RpcBlockConfig;
 use solana_sdk::slot_history::Slot;
-use std::{ops::Range, sync::Arc};
+use std::ops::RangeInclusive;
+use std::sync::Arc;
 
 #[async_trait]
 pub trait BlockStorageInterface: Send + Sync {
     // will save a block
-    async fn save(&self, block: ProducedBlock) -> Result<()>;
+    // TODO: slot might change for a sig if the block gets confirmed/finalized
+    async fn save(&self, block: &ProducedBlock) -> Result<()>;
```

Comment:
I have consistently changed all parameters to refs to avoid cloning. but maybe I should treat `.save` differently. is it that, what you are advocating for.